app/distributions/api.py

```diff
@@ -0,0 +1,149 @@
+            en=model.description_en,
+            it=model.description_it,
+            rm=model.description_rm,
+        ),
+        provider_id=str(model.provider.id),
+    )
+    return response
+
+
+@router.get("attributions/{attribution_id}", response={200: AttributionSchema}, exclude_none=True)
+def attribution(request: HttpRequest, attribution_id: int, lang: LanguageCode | None = None):
+    """
+    Get the attribution with the given ID, return translatable fields in the given language.
+
+    Example: If German ("de") is set as the language of choice, fields "name" and
+             "description" would take the value of the corresponding translation.
+
+                {
+                    "id": "1",
+                    "name": "German",
+                    "name_translations": {
+                        "de": "German",
+                        "fr": "French",
+                        "en": "English",
+                        "it": "Italian",
+                        "rm": "Romansh",
+                    },
+                    "description": "DE",
+                    "description_translations": {
+                        "de": "DE",
+                        "fr": "FR",
+                        "en": "EN",
+                        "it": "IT",
+                        "rm": "RM",
+                    },
+                    provider_id: 123
+                }
+
+    The language can be set via
+
+        1. Query parameter. For example: "lang=de"
+        2. Header "Accept-Language". For example: "Accept-Language: de"
+
+    To consider for the language setting:
+
+        - If both query param and header are set, the query param is taken.
+        - The valid languages are: "de", "fr", "en", "it", "rm"
+        - If no language is specified, English is taken as the default.
+
+    To consider for the header "Accept-Language":
+
+        - For multiple languages, the first valid language is taken, so q-factors
+          are ignored (Example: "de;q=0.7, rm;q=0.8" --> "de")
+        - Subtags in the header are ignored. So "en-US" is interpreted as "en".
+        - Wildcards ("*") are ignored.
+    """
+    model = get_object_or_404(Attribution, id=attribution_id)
+    lang_to_use = get_language(lang, request.headers)
+    response = attribution_to_response(model, lang_to_use)
+    return response
+
+
+@router.get("attributions", response={200: AttributionListSchema}, exclude_none=True)
+def attributions(request: HttpRequest, lang: LanguageCode | None = None):
+    """
+    Get all attributions, return translatable fields in the given language.
+
+    For more details on how individual attributions are returned, see the
+    corresponding endpoint for a specific attribution.
+    """
+    models = Attribution.objects.order_by("id").all()
+    lang_to_use = get_language(lang, request.headers)
+
+    responses = [attribution_to_response(model, lang_to_use) for model in models]
+    return AttributionListSchema(items=responses)
```

Comment:
Is it necessary to explicitly call the Schema-Serializer? From the [docs](https://django-ninja.dev/tutorial/other/crud/#list-of-objects) it seems that you could only return the list and let Ninja handling the Serialization?

Reply:
Good point, I originally wanted to do that. The thing is that we have it wrapped into the `items=[...]` keyword, so I need a Schema that spells this out.

Do you know what's the benefit of this `items`? It's in every example of [the API specs](https://2zvidp4qyh.apidog.io/api-9150424) but I am not sure what is its purpose.

Reply:
I guess the way would be to return a dict like so

```
return {"items": responses}
```

Since you specify `AttributionListSchema` in the function decorator my assumption is that Ninja takes care of actually schema-ify the data afterwards.

Reply:
Ok, so I would suggest to keep the schema because
1. We might want to add more fields than just "items".
2. It validates that we give it a list. Otherwise we might give it some other other iterable by accident, like a set.

Concerns with that @schtibe?

Reply:
Not sure if we're still talking about the same thing 😅 I made a PR in your PR to demonstrate what I mean:
https://github.com/geoadmin/service-control/pull/27

Reply:
Ok, discussed this in person and left a [conclusion in the corresponding PR](https://github.com/geoadmin/service-control/pull/27#issuecomment-2399904098). I apply the relevant changes here.